### ISSUE TITLE
Admin doesn't list products that have different language than admin user language

### ISSUE THREAD

**Issue by floriangosse** [Sylius/Sylius#12145]
**Sylius version affected**: 1.8.5

**Description**  
I have an admin user that has the language `en_US` set as their interface language. All the products in the system are set up for the German (`de_DE`) language. The mentioned user can't see the products in the product listing.

**Steps to reproduce**  
1. Setup Sylius as described in the documentation
1. Don't add sample data
1. Login with created admin user
1. Setup channel, zones, etc. ...
1. Setup language "German" as the only language for the shop
1. Create a product
1. Change the language of the admin user to another language than German
1. Open the product listing

**Possible Solution**  
Ideally, all products are listed independently of the admin user's configured language.

**Comment by PACMAN49** [comment]
i think it s relate to this issue #11521 
i reproduce the issue with french language as well

**Comment by zangra-dev** [comment]
The problem come from the Sylius\Bundle\CoreBundle\Doctrine\ORM\ProductRepository => createListQueryBuilder
Inside this function there is an innerJoin on translation
```
$queryBuilder = $this->createQueryBuilder('o')
    ->addSelect('translation')
    ->innerJoin('o.translations', 'translation', 'WITH', 'translation.locale = :locale')
    ->setParameter('locale', $locale)
;
```

With the overriding of this function and delete the inner join the product is shown with the default translation (wich is mandatory to create the product)

**Comment by zangra-dev** [comment]
Comment the innerJoin is not really the solution (because of the search's crash). I try to change it with a leftJoin but, because of the empty translation a blank cell is shown. I added another leftJoin to force to get the EN translation.

The product is correctly display in the product admin index page with the EN translation and when I do a search, there is no more crash

Here is the code used for the createListQueryBuilder
```PHP
public function createListQueryBuilder(string $locale, $taxonId = null): QueryBuilder
{
    $queryBuilder = $this->createQueryBuilder('o')
        ->addSelect('translation')
        ->leftJoin('o.translations', 'translation', 'WITH', 'translation.locale = :locale')
        ->setParameter('locale', $locale)

        ->addSelect('translation')
        ->leftJoin('o.translations', 'translationEN', 'WITH', 'translationEN.locale = :localeEN')
        ->setParameter('localeEN', 'en_US')
    ;

    if (null !== $taxonId) {
        $queryBuilder
            ->innerJoin('o.productTaxons', 'productTaxon')
            ->andWhere('productTaxon.taxon = :taxonId')
            ->setParameter('taxonId', $taxonId)
        ;
    }

    return $queryBuilder;
}
```

**Comment by NoResponseMate** [comment]
Hi @floriangosse 
I'm closing this as it's been fixed in #13724

Cheers 🍻
